chemprop/features/featurization.py

```diff
@@ -166,7 +218,19 @@ def __init__(self, mol: Union[str, Chem.Mol],
         """
         # Convert SMILES to RDKit molecule if necessary
         if type(mol) == str:
-            mol = Chem.MolFromSmiles(mol)
+            if REACTION_MODE:
```

Comment:
I might be mistaken, but how does this check work? Either reaction mode hasn't been set so this is None and it errors, or it's a string and passes. 

Should this be `if REACTION_MODE is not None`?

Reply:
If reaction mode hasn't been set, it is None (otherwise a string). The boolean value of None is False, and True for a string (or other objects that are not None), so the check works out fine - but I have to admit it is rather ugly ;) It is actually an artefact, since I started out with `REACTION_MODE` being Boolean. I will correct this to `if REACTION_MODE is not None` to make the code more readable.

Reply:
Relying on falsyness of `None` is generally not a problem. Using `is not None` is more specific and is better in situations where other falsy values are possible and need to pass the check (e.g. `0`, `''`).

Reply:
Since `REACTION_MODE` can only be `None`, `'Reac_Diff'`, `'Prod_Diff'` or `'Reac_Prod'`, I guess there is nothing that could currently go wrong  - but, to avoid bugs in future code, it might be easier to read/understand to just say `if REACTION_MODE != None`, so I am happy to change that!

Reply:
Sounds good. Also, the syntax in your previous comment was better; comparisons against `None` should be done using `is` rather than `=`.

Reply:
I thought it was more unstable than that. In that case I have no problem here.

Reply:
Due to changes in the global variables REACTION and REACTION MODE, as well as a helper function to make RDKit molecules from strings, this should be better readable now